source/Options.cpp

```diff
@@ -147,336 +147,176 @@ Options::Options(
       export_origins_mode_(export_origins_mode),
       propagate_across_arguments_(propagate_across_arguments) {}
 
-Options::Options(const boost::program_options::variables_map& variables) {
-  system_jar_paths_ = parse_paths_list(
-      variables["system-jar-paths"].as<std::string>(),
-      std::nullopt,
-      /* check exist */ false);
+Options::Options(const Json::Value &json){
+    system_jar_paths_ = parse_paths_list(
+        json["system-jar-paths"].asString(),
+        std::nullopt,
+        /* check exist */ false);
```

Comment:
using `json["system-jar-paths"].asString()` converts the list to a string and `parse_paths_list` converts the string back to a list. This is wasteful. We should write a `JsonValidation::string_list()` function. See JsonValidation.cpp for examples on how to parse a list of json elements.

Reply:
Here We should use `parse_paths_list ` as it handles if there is a directory if extracts the files on it

Reply:
I think what @arthaud means is: 
- Now that we are passing the arguments as a json file, we should take arguments that take a list as a json array and *not* a `;` separated string like before. 
- For eg. we want to pass 2 paths to `system-jar-paths` argument: `path/to/1` and `path/to/2`.
- You are creating the options json as: 
```
  "system-jar-paths": "path/to/1;path/to/2"
```
- In `shim.py`, you already split this argument by `;` to check if the paths are valid. So no need to join the list back into a string anymore since we are creating a json file. 
  - See https://github.com/ZeyadTarekk/mariana-trench/blob/3401b4818bcc14c2ea8f3d74c89271540b539860/shim/shim.py#L69-L75. You'll have to update that and other similar helpers in `shim.py` accordingly.

- Instead of the string, we should create a json array for these keys like:
```
"system-jar-paths": ["path/1", "path/2"]
```
- Then, here you can add a `JsonValidation::string_list()` method to parse out this json array 
- `parse_paths_list()` currently takes a string and splits it on `;`. This then will not be necessary.